packages/ts-client/src/reconnection.ts

```diff
@@ -96,15 +96,18 @@ export class ReconnectManager<PeerMetadata, ServerMetadata> {
     return this.status === 'reconnecting';
   }
 
-  public reset(initialMetadata: PeerMetadata) {
-    this.initialMetadata = initialMetadata;
+  public reset(initialPeerMetadata: PeerMetadata) {
+    this.initialPeerMetadata = initialPeerMetadata;
     this.reconnectAttempt = 0;
     if (this.reconnectTimeoutId) clearTimeout(this.reconnectTimeoutId);
     this.reconnectTimeoutId = null;
   }
 
   private getLastPeerMetadata(): PeerMetadata | undefined {
-    return this.lastLocalEndpoint?.metadata as PeerMetadata;
+    const endpointMetadata = this.lastLocalEndpoint?.metadata as
+      | { peer: PeerMetadata; server: ServerMetadata }
```

Comment:
```suggestion
      | Metadata<PeerMetadata, ServerMetadata>
```

https://github.com/fishjam-cloud/web-client-sdk/blob/af10e28c81475a654848b55658f60509693cf641/packages/ts-client/src/types.ts#L30

Reply:
yup! that's how it should be